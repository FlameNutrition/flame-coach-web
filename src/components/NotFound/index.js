import {
  Box, Container, makeStyles, Typography
} from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import PageNotFound from '../../images/404';
import { useIsMobile } from '../../utils/mediaUtil';

const useStyles = makeStyles(() => ({
  root: {
    height: '100%',
    paddingLeft: '15px',
    paddingRight: '15px'
  },
  imageLayout: {
    marginTop: '10px',
    width: '100%'
  },
  imageMobile: {
    height: 'auto',
    width: '50%'
  },
  image: {
    width: '300pt',
    height: '300pt'
  }
}));

const NotFound = ({ title, submessage }) => {
  const classes = useStyles();
  const isMobile = useIsMobile();

  return (
    <Box
      display="flex"
      flexDirection="column"
      className={classes.root}
      justifyContent="center"
    >
      <Container maxWidth="md">
        <Typography
          align="center"
          color="textPrimary"
          variant="h1"
        >
          {title}
        </Typography>
        <Typography
          align="center"
          color="textPrimary"
          variant="subtitle2"
        >
          {submessage}
        </Typography>
        <Box
          display="flex"
          justifyContent="center"
          className={classes.imageLayout}
        >
          <PageNotFound className={isMobile ? classes.imageMobile : classes.image} viewBox="0 0 496 496" />
        </Box>
      </Container>
    </Box>
  );
};

NotFound.propTypes = {
  title: PropTypes.string.isRequired,
  submessage: PropTypes.string.isRequired
};

NotFound.defaultProps = {};

export default NotFound;
