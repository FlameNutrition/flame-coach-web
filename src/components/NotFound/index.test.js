import React from 'react';
import { render } from '../../testing/test-utils';
import NotFound from './index';

describe('<NotFound/ >', () => {
  it('create component', () => {
    const { container } = render(
      <NotFound
        title="The page you are looking for isn’t here"
        submessage="You either tried some shady route or you came here by mistake. Whichever it is, try using the navigation"
      />
    );

    expect(container).toMatchSnapshot();
    expect(container.querySelector('h1')).toHaveTextContent('The page you are looking for isn’t here');
    expect(container.querySelector('h6')).toHaveTextContent('You either tried some shady route or you came here by mistake. Whichever it is, try using the navigation');
    expect(container.querySelector('svg')).not.toBeNull();
  });
});
