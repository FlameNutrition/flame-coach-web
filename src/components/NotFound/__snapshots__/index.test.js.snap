// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/ > create component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-root-1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
      >
        The page you are looking for isnâ€™t here
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
      >
        You either tried some shady route or you came here by mistake. Whichever it is, try using the navigation
      </h6>
      <div
        class="MuiBox-root MuiBox-root-6 makeStyles-imageLayout-2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-image-4"
          focusable="false"
          viewBox="0 0 496 496"
        >
          <g>
            <path
              d="M84.446,317.403H13.082C5.649,317.403,0,312.348,0,303.724c0-2.378,39.246-146.295,39.246-146.295 c3.568-7.434,10.11-10.11,16.354-10.11c6.839,0,23.99,4.614,23.99,13.536c0,1.487,0.073,4.603-0.82,6.386L49.655,281.423h34.791 v-31.221c0-8.622,9.812-12.191,19.327-12.191c9.814,0,19.328,3.568,19.328,12.191v31.221h13.975 c8.029,0,12.192,8.921,12.192,18.138c0,8.92-5.949,17.842-12.192,17.842h-13.975v35.087c0,8.029-9.515,12.191-19.328,12.191 c-9.515,0-19.327-4.163-19.327-12.191V317.403z"
              fill="#4f5eb3"
            />
            <path
              d="M390.515,199.571c6.446-23.746,11.462-42.142,11.462-42.142c3.568-7.434,10.11-10.11,16.354-10.11 c6.839,0,23.99,4.614,23.99,13.536c0,1.487,0.073,4.603-0.82,6.386l-29.114,114.182h34.791v-31.221 c0-8.622,9.812-12.191,19.327-12.191c9.814,0,19.328,3.568,19.328,12.191v31.221h13.975c8.029,0,12.192,8.921,12.192,18.138 c0,8.92-5.949,17.842-12.192,17.842h-13.975v35.087c0,8.029-9.515,12.191-19.328,12.191c-9.515,0-19.327-4.163-19.327-12.191 v-35.087h-71.365c-7.433,0-13.082-5.055-13.082-13.679c0-1.191,9.84-37.869,19.666-74.197L390.515,199.571z"
              fill="#4f5eb3"
            />
          </g>
          <circle
            cx="256.009"
            cy="263.073"
            fill="#fc8868"
            r="88.481"
          />
          <path
            d="M267.088,263.076l30.777-30.777c3.06-3.06,3.06-8.023,0-11.082c-3.062-3.06-8.023-3.06-11.083,0 l-30.777,30.777l-30.777-30.777c-3.062-3.06-8.023-3.06-11.083,0c-3.06,3.06-3.06,8.023,0,11.082l30.777,30.777l-30.777,30.777 c-3.06,3.06-3.06,8.023,0,11.082c1.531,1.53,3.537,2.296,5.542,2.296s4.011-0.765,5.542-2.296l30.777-30.777l30.777,30.777 c1.531,1.53,3.537,2.296,5.542,2.296s4.011-0.765,5.542-2.296c3.06-3.06,3.06-8.023,0-11.082L267.088,263.076z"
            fill="#FFFFFF"
          />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
